recipes/repl-builtin-modules/src/workflow.ts

```diff
@@ -0,0 +1,309 @@
+                    rule: { kind: "pair_pattern" }
+                });
+
+				// Check if only builtin properties are destructured
+				const builtinShorthandCount = properties.filter(p =>
+					containsBuiltinProperties(p.text())
+				).length;
+
+                const builtinPairCount = pairProperties.filter(p =>
+                    containsBuiltinProperties(p.text())
+                ).length;
+
+                const totalBuiltinCount = builtinShorthandCount + builtinPairCount;
+                const totalCount = properties.length + pairProperties.length;
+
+                const isOnlyBuiltin = totalCount > 0 && totalBuiltinCount === totalCount;
+
+                if (isOnlyBuiltin) {
+                    // Replace entire require statement
+                    updateModuleSpecifier(statement);
+
+                    if (originalText.includes("_builtinLibs")) {
+                        const newText = originalText.replace("_builtinLibs", "builtinModules");
+                        edits.push(objectPattern.replace(newText));
+                        replaceStandaloneBuiltinLibsReferences();
+                    }
+                } else {
+                    // Split into two statements
+                    const propertiesToKeep: string[] = [];
+                    const builtinAliases: string[] = []; // Change to array to handle multiple aliases
+
+                    for (const prop of properties) {
+                        const propText = prop.text();
+                        if (propText !== "builtinModules" && propText !== "_builtinLibs") {
+                            propertiesToKeep.push(propText);
+                        }
+                    }
+
+                    for (const prop of pairProperties) {
+                        const propText = prop.text();
+
+                        if (containsBuiltinProperties(propText)) {
+                            // Extract alias from pair pattern like "builtinModules: alias" or "_builtinLibs: alias"
+                            const keyNode = prop.find({ rule: { kind: "property_identifier" } });
+                            const valueNode = prop.find({ rule: { kind: "identifier" } });
+
+							if (
+								keyNode &&
+								valueNode &&
+								containsBuiltinProperties(keyNode.text())
+							) {
+								builtinAliases.push(valueNode.text());
+							}
+                        } else {
+                            propertiesToKeep.push(propText);
+                        }
+                    }
+
+                    if (propertiesToKeep.length > 0) {
+                        const variableDeclaration = statement.parent();
+                        const moduleSpecifier = statement.find({ rule: { kind: "string" } });
+
+                        if (variableDeclaration && moduleSpecifier) {
+                            const currentModule = moduleSpecifier.text();
+                            const newModule = "'node:module'";
```

Comment:
In a previous comment, I said that const can be removed as not used in other place, looking now, it can be created in a higher scope (begin of transform function) to be reused through the code.

```suggestion
                            const newModule = "'node:module'";
```